### ISSUE TITLE
Unknown type: :erlang.monitor_option/0 on Elixir 1.13.1 with OTP 23.3

### ISSUE THREAD

**Issue by thomas9911** [elixir-lang/elixir#11516]
### Environment

* Elixir & Erlang/OTP versions (elixir --version): 
Elixir: 1.13.1
Erlang/OTP: 23.3

* Operating system: 
Ubuntu 18.04  (my machine)
Alpine 1.15.0  (gitlab ci)

### Current behavior

While running Dialyzer with my configuration of Elixir and Erlang I get this error message.

```
:0:unknown_type Unknown type: :erlang.monitor_option/0.
done (warnings were emitted)
Halting VM with exit status 2
```

I dove in a bit and found that this type was added to Erlang 24 but this type is used in the Process spawn_opts of elixr 1.13.1 https://github.com/elixir-lang/elixir/blob/8b2cd967615e9bf9b59ee97ae17ad0bc27928bf7/lib/elixir/lib/process.ex#L383

I looked at the compatibily list of Elixir and it says that 1.13 should be compatible with OTP 22-24

### Expected behavior

Dialyzer should not error

### Workaround

I think this is 'workaround'able to add this error message to my dialyzer ignore file

**Comment by josevalim** [comment]
Fixed in main and v1.13.

**Comment by michallepicki** [comment]
The whole `spawn_opt` could be just a reference to Erlang type and it would always behave as expected then, but I believe those are inlined so that they are visible in Elixir documentation. Hopefully with the extra type inlined as well Dialyzer won't complain for you in the older Erlang version, even though it's incompatible :)
